internal/dao/stubs/pubkey_dao.go

```diff
@@ -19,36 +20,52 @@ func init() {
 	dao.GetPubkeyDao = getPubkeyDao
 }
 
-type ContextReadError struct{}
+type contextReadError struct{}
 
-func (m *ContextReadError) Error() string {
+func (m *contextReadError) Error() string {
 	return "failed to find or convert dao stored in testing context"
 }
 
 func WithPubkeyDao(parent context.Context, init_store []*models.Pubkey) context.Context {
-	ctx := context.WithValue(parent, pubkeyCtxKey, PubkeyDaoStub{init_store})
+	ctx := context.WithValue(parent, pubkeyCtxKey, &PubkeyDaoStub{init_store})
 	return ctx
 }
 
+func PubkeyStubCount(ctx context.Context) (int, error) {
+	pkdao, err := getPubkeyDaoStub(ctx)
+	if err != nil {
+		return 0, err
+	}
+	return len(pkdao.store), nil
+}
+
 func getPubkeyDao(ctx context.Context) (dao.PubkeyDao, error) {
+	return getPubkeyDaoStub(ctx)
+}
+
+func getPubkeyDaoStub(ctx context.Context) (*PubkeyDaoStub, error) {
 	var err error
-	pkdao, ok := ctx.Value(pubkeyCtxKey).(dao.PubkeyDao)
+	pkdao, ok := ctx.Value(pubkeyCtxKey).(*PubkeyDaoStub)
 	if !ok {
-		err = &ContextReadError{}
+		err = &contextReadError{}
 	}
-
 	return pkdao, err
 }
 
-func (PubkeyDaoStub) Create(ctx context.Context, pk *models.Pubkey) error { return nil }
-func (PubkeyDaoStub) CreateWithResource(ctx context.Context, pk *models.Pubkey, pkr *models.PubkeyResource) error {
+func (mock *PubkeyDaoStub) Create(ctx context.Context, pk *models.Pubkey) error {
+	log.Print(mock.store)
+	mock.store = append(mock.store, pk)
+	log.Print(mock.store)
+	return nil
```

Comment:
log from where? I am bit confused, anyway, I would not perform any logging in tests, they either pass or fail and should be small enough that they should not require logging. My concern is then we can end up with too many logs (remember Foreman test.log size :-)?

Reply:
it's totally a debug leftover :( I need to lear debug in a way linter will yell at me :)